src/Aspire.Hosting.Azure.Provisioning/Provisioners/AISearchProvisioner.cs

```diff
@@ -0,0 +1,71 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using Azure;
+using Azure.ResourceManager.Search;
+using Microsoft.Extensions.Configuration;
+using Microsoft.Extensions.Logging;
+
+namespace Aspire.Hosting.Azure.Provisioning.Provisioners;
+
+internal sealed class AISearchProvisioner(ILogger<AISearchProvisioner> logger) : AzureResourceProvisioner<AzureAISearchResource>
+{
+    public override bool ConfigureResource(IConfiguration configuration, AzureAISearchResource resource)
+    {
+        if (configuration.GetConnectionString(resource.Name) is string connectionString)
+        {
+            resource.ConnectionString = connectionString;
+            return true;
+        }
+
+        return false;
+    }
+
+    public override async Task GetOrCreateResourceAsync(
+        AzureAISearchResource resource,
+        ProvisioningContext context,
+        CancellationToken cancellationToken)
+    {
+        context.ResourceMap.TryGetValue(resource.Name, out var azureResource);
+
+        if (azureResource is not null && azureResource is not SearchServiceResource)
+        {
+            logger.LogWarning("Resource {resourceName} is not an AI Search resource. Deleting it.", resource.Name);
+
+            await context.ArmClient.GetGenericResource(azureResource.Id).DeleteAsync(WaitUntil.Started, cancellationToken).ConfigureAwait(false);
+        }
+
+        var aiSearchResource = azureResource as SearchServiceResource;
+
+        if (aiSearchResource is null)
+        {
+            var aiSearchName = Guid.NewGuid().ToString().Replace("-", string.Empty);
+
+            var parameters = new SearchServiceData(context.Location);
+
+            logger.LogInformation("Creating AI Search {aiSearchName} in {location}...", aiSearchName, context.Location);
+
+            var aiSearchCreateOperation = await context.ResourceGroup
+                .GetSearchServices().CreateOrUpdateAsync(WaitUntil.Completed, aiSearchName, parameters, cancellationToken: cancellationToken).ConfigureAwait(false);
+            aiSearchResource = aiSearchCreateOperation.Value;
+
+            logger.LogInformation("AI Search {aiSearchName} created.", aiSearchResource.Data.Name);
+        }
+
+        // SearchServiceResource doesn't have an "Endpoint" property
+        resource.ConnectionString = $"https://{aiSearchResource.Data.Name.ToLowerInvariant()}.search.windows.net";
```

Comment:
Do we know that the current `context.ResourceGroup` is pointing to public Azure?  (Is anyone testing Aspire provisioning against federated Azure?)

Reply:
Looks like we're using [`AzureLocation`](https://learn.microsoft.com/dotnet/api/azure.core.azurelocation?view=azure-dotnet)  [here](https://github.com/dotnet/aspire/blob/813fc19c3acf1a644f23157f608a40a2f4ce829f/src/Aspire.Hosting.Azure.Provisioning/Provisioners/AzureProvisioner.cs#L77). I'm not sure/familiar with this, but based on the its properties, looks like they all represent public Azure(?)

Reply:
If the URL doesn't come back from the resource provider, then we have no choice to but to build URLs. Does the resource have a URI property?

Reply:
Nope, it doesn't

Reply:
https://learn.microsoft.com/en-us/azure/ai-services/create-account-bicep?tabs=CLI

I find it so strange that these samples never show how to connect to the resource... (connection string, keys, urls etc).

Reply:
`AzureLocation` will allow Federated Azure as well (you'll see Gov, Germany, and China locations in there).  You'd need to set your endpoint accordingly if not pointing to public Azure though.

Search is a very old service.  If an RP predates Federated Azure or Azure Stack, it's more likely to assume you'd be happy hardcoding `$"https://{acct}.{service}.windows.net"` everywhere.

We should brainstorm whether this is something we can handle in the CDK and try to contain all the magic URI templates in one place while we wait for services to add the knob.